docs/proposal-config-changes.md

```diff
@@ -0,0 +1,731 @@
+# Proposal - changes to global and workflow configs
+
+These notes mainly address [cylc-flow #3422](https://github.com/cylc/cylc-flow/issues/3422).
+
+## General points
+
+### Terminology
+   * We should remove all references to `suite` in the settings (partly because
+     we're trying stop referring to "suites" and partly because some affected
+     settings are actually referring to the workflow server which is unclear).
+     * Note: "suite" is referred to through-out the code, tests and
+       documentation.
+   * Do we want to use `server` as shorthand for `workflow server`?
+     * Presumably there is no confusion with `UI server` since this will have a
+       separate config file?
+     * Using `server` for now in this proposal.
+     * See
+       [discussion](https://github.com/cylc/cylc-flow/pull/3348#discussion_r332985001).
+
+### Deprecation
+   * The global config will be changing drastically for the platforms support.
+     Furthermore, it doesn't affect most end users.
+     * Proposal: Don't support any deprecated or obsolete settings in the global
+       config.
+   * For the workflow config the code to support deprecated and obsolete settings
+     isn't complicated so there is no big overhead for us. On the other hand we
+     don't want users using old syntax for too long. Some settings only became
+     deprecated or obsolete very recently - it seems harsh to remove support for
+     these.
+     * Proposal: Remove everything from 7.6.0 (Feb 2018) and before. That will
+       just leaves things from 7.8.x.
+
+### Config file naming
+
+See [cylc-flow #3170](https://github.com/cylc/cylc-flow/issues/3170).
+   * Suite: Currently `suite.rc` (specified in
+     [suite_files.py](https://github.com/cylc/cylc-flow/blob/master/cylc/flow/suite_files.py#L123)).
+     Since we're trying to move away from referring to "suites" we should change
+     this (with backwards compatibility).
+     * Proposal: `flow.rc` or `cylc-flow.rc`
+   * Global
+     * Currently `flow.rc` (specified in
+       [globalcfg.py](https://github.com/cylc/cylc-flow/blob/master/cylc/flow/cfgspec/globalcfg.py#L325)).
+       * Proposal: `config.rc`?
+     * Read from `/etc/cylc/flow/8.0a1/flow.rc` and
+       `~/.cylc/cylc/flow/8.0a1/flow.rc` unless `$CYLC_CONF_PATH` set in which
+       case it is read only from this path.
+     * We need the ability to override where the site config file is read from,
```

Comment:
This sounds like it would make testing easier - I like the idea of being able to set both the site and user conf paths separately.

Reply:
(updated post merge)